server/server_test.go

```diff
@@ -287,3 +294,89 @@ func TestMiddlewareLogging(t *testing.T) {
 	require.NoError(t, err)
 	http.DefaultClient.Do(req)
 }
+
+func TestTLSServer(t *testing.T) {
+	var level logging.Level
+	level.Set("info")
+
+	cmd := exec.Command("bash", "certs/genCerts.sh", "certs", "1")
+	err := cmd.Run()
+	require.NoError(t, err)
+
+	cfg := Config{
+		HTTPListenAddress: "localhost",
+		HTTPListenPort:    9193,
+		HTTPTLSConfig: node_https.TLSStruct{
+			TLSCertPath: "certs/server.crt",
+			TLSKeyPath:  "certs/server.key",
+			ClientAuth:  "RequireAndVerifyClientCert",
+			ClientCAs:   "certs/root.crt",
+		},
+		GRPCTLSConfig: node_https.TLSStruct{
+			TLSCertPath: "certs/server.crt",
+			TLSKeyPath:  "certs/server.key",
+			ClientAuth:  "VerifyClientCertIfGiven",
+			ClientCAs:   "certs/root.crt",
+		},
+		MetricsNamespace:  "testing_tls",
+		GRPCListenAddress: "localhost",
+		GRPCListenPort:    9194,
+	}
+	server, err := New(cfg)
+	require.NoError(t, err)
+
+	server.HTTP.HandleFunc("/testhttps", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("Hello World!"))
+	})
+
+	fakeServer := FakeServer{}
+	RegisterFakeServerServer(server.GRPC, fakeServer)
+
+	go server.Run()
+	defer server.Shutdown()
+
+	clientCert, err := tls.LoadX509KeyPair("certs/client.crt", "certs/client.key")
+	if err != nil {
+		log.Warnf("error loading cert %s or key %s, tls disabled", "certs/client.crt", "certs/client.key")
```

Comment:
what does it mean for the test if this warning fires?

Reply:
Changing this to `require.NoError()`